### PR TITLE
Remediate and enforce cli-path-validation CI rail

### PR DESCRIPTION
**Pull request by curdriceaurora**
This PR resolves the technical debt for the `cli-path-validation` CI rail, tunes the AST-based validation scanner to filter out renderer/helper functions and annotation string false positives, adds a comprehensive test suite for the scanner, and promotes the rail to enforce mode.

Closes #1365

## Summary by CodeRabbit

* **New Features**
  * CLI commands now resolve and validate user-provided paths before using them (token save, API key output, doctor scanning, and recover journal handling).

* **Bug Fixes**
  * Added stricter checks to ensure resolved paths are regular files (or directories where required), with clearer errors for invalid inputs.

* **CI / Tests**
  * The `cli-path-validation` guardrail is now enforced in CI.
  * Expanded guardrail coverage and added CLI/integration tests for rejecting directory/non-file outputs, plus additional memory profiler tests across Linux/Darwin.

### CODE REVIEW

File: scripts/ci/guardrails/check_cli_path_validation.py [hunk 1]
```diff
@@ -19,6 +19,45 @@
...
+
+    def visit_Name(self, node: ast.Name) -> None:
+        if node.id == "Path":
+            self.has_path = True
+        self.generic_visit(node)
+
+    def visit_Attribute(self, node: ast.Attribute) -> None:
+        if node.attr == "Path":
+            self.has_path = True
+        self.generic_visit(node)
```
[the remark this answers is not in the document]
Comment (curdriceaurora) [reply scripts/ci/guardrails/check_cli_path_validation.py:69]: Accepted. I updated the checker so `Annotated[...]` only inspects the declared type expression and ignores metadata, which removes false positives from metadata references like `path_type=Path`. I also added nested coverage to ensure this behavior stays correct.

File: src/file_organizer/cli/api_keys.py [hunk 2]
```diff
@@ -36,9 +36,10 @@ def generate_key(
 ) -> None:
     """Generate a secure API key and print its bcrypt hash."""
     from file_organizer.api.api_keys import _write_key, generate_api_key, hash_api_key
+    from file_organizer.cli.path_validation import resolve_cli_path
 
     # Resolve output path
-    resolved_output = output.expanduser()
+    resolved_output = resolve_cli_path(output, must_exist=False, must_be_dir=False)
     api_key = generate_api_key(prefix=prefix)
     _write_key(resolved_output, api_key)
```
[the remark this answers is not in the document]
Comment (curdriceaurora) [reply src/file_organizer/cli/api_keys.py:46]: Partially accepted. The CLI already rejects an existing directory for `--output` with `BadParameter`, so no additional production change was needed there. I added a regression CLI test to lock this behavior in place.

File: tests/ci/test_check_cli_path_validation.py [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+    src.write_text(
+        """
+@app.command()
+def run(directory: Path) -> None:  # noqa: cli-path-validation
+    pass
+""",
+        encoding="utf-8",
+    )
+    violations = checker.check_file(src)
+    assert len(violations) == 0
```
[the remark this answers is not in the document]
Comment (curdriceaurora) [reply tests/ci/test_check_cli_path_validation.py:131]: Partially accepted. The test was already aligned to `# copilot: wontfix` and the checker already accepted that suppression format. I kept that behavior and extended checker coverage for related `Annotated` edge cases.

File: tests/integration/test_optimization_core.py [hunk 4]
```diff
@@ -203,6 +203,107 @@ def test_get_top_objects_returns_sorted_list(self) -> None:
...
+
+    def test_get_rss_resource_import_error(self) -> None:
+        from file_organizer.optimization.memory_profiler import MemoryProfiler
+
+        with (
+            patch("builtins.open", side_effect=FileNotFoundError),
+            patch("sys.platform", "linux"),
+            patch("resource.getrusage", side_effect=ImportError),
+        ):
+            rss = MemoryProfiler._get_rss()
```
[the remark this answers is not in the document]
Comment (curdriceaurora) [reply tests/integration/test_optimization_core.py:257]: Accepted. I switched both tests to simulate a missing `resource` module via `patch.dict(sys.modules, {"resource": None})`, which exercises the intended import-failure path. I re-ran the targeted memory profiler tests and they pass.

**Comment by curdriceaurora** [comment]
Approved
